notifications_utils/template.py

```diff
@@ -49,7 +50,7 @@
 ))
 
 
-class Template():
+class Template(ABC):
```

Comment:
I have:
- made `LetterImageTemplate` inherit from `BaseLetterTemplate` in 79b9fef
- made a `BaseSMSTemplate` in 6d1ebe6

I didn’t:
- make `LetterPrintTemplate` inherit from `BaseLetterTemplate` (it shares too much with `LetterPreviewTemplate`, and extracting the shared stuff out into another abstract class or mixin felt like it would be over-egging things)

Reply:
So the updated list is:
```
class Template(ABC)

class BaseSMSTemplate(Template)

class SMSMessageTemplate(BaseSMSTemplate)

class SMSPreviewTemplate(BaseSMSTemplate)

class SubjectMixin()

class BaseEmailTemplate(SubjectMixin, Template)

class PlainTextEmailTemplate(BaseEmailTemplate)

class HTMLEmailTemplate(BaseEmailTemplate)

class EmailPreviewTemplate(BaseEmailTemplate)

class BaseLetterTemplate(SubjectMixin, Template)

class LetterPreviewTemplate(BaseLetterTemplate)

class LetterPrintTemplate(LetterPreviewTemplate)

class LetterImageTemplate(BaseLetterTemplate)
```

Reply:
Very optional (so optional that even I don't think I'd do it myself): Would be nice if all the email classes started with the word `Email`, like the letter and SMS classes do. Probably not worth the effort though, I just like things to look pretty ;)

Reply:
Yeah, `EmailHTMLTemplate` or `EmailHTMLEmailTemplate` don’t feel great either 😕